internal/controller/kubeobjects/velero/requests.go

```diff
@@ -172,7 +173,7 @@ func (r RequestsManager) RecoverRequestsDelete(
 
 func (RequestsManager) RecoverRequestCreate(
 	ctx context.Context,
-	writer client.Writer,
+	writer client.Client,
```

Comment:
(nit) The variable name can also just be client, instead of writer. That helps when reading the code that a client is involved and not necessarily a writer client. (here and other functions where the signature is changed)